ios/Exponent/Kernel/Services/Notifications/EXApiV2Client+EXRemoteNotifications.m

```diff
@@ -40,16 +40,23 @@ - (NSURLSessionTask *)updateDeviceToken:(NSData *)deviceToken completionHandler:
 
 
 - (NSURLSessionTask *)getExpoPushTokenForExperience:(NSString *)experienceStableLegacyId
+                                       easProjectId:(NSString *)easProjectId
                                         deviceToken:(NSData *)deviceToken
                                   completionHandler:(void (^)(NSString * _Nullable, NSError * _Nullable))handler
 {
   NSMutableDictionary *arguments = [NSMutableDictionary dictionaryWithDictionary:@{
     @"deviceId": [EXKernel deviceInstallationUUID],
-    @"experienceId": experienceStableLegacyId,
     @"appId": NSBundle.mainBundle.bundleIdentifier,
     @"deviceToken": deviceToken.apnsTokenString,
     @"type": @"apns",
   }];
+  
+  if (easProjectId != nil) {
+    arguments[@"projectId"] = easProjectId;
+  } else if (experienceStableLegacyId != nil) {
+    arguments[@"experienceId"] = experienceStableLegacyId;
+  }
```

Comment:
if neither is provided, should we pass an error to the completion handler?

Reply:
I did this on android, so yep, I should probably do it here as well. The endpoint would throw as well but it is good to preempt it.